roxie/udplib/udptrr.cpp

```diff
@@ -136,44 +148,74 @@ class CReceiveManager : implements IReceiveManager, public CInterface
             timeouts = 0;
         }
 
-        inline bool retryOnTimeout()
+        bool canSendAny() const
         {
-            ++timeouts;
-            if (udpTraceLevel)
+            // We can send some if (a) the first available new packet is less than TRACKER_BITS above the first unreceived packet or
+            // (b) we are assuming arrival in order, and there are some marked seen that are > first unseen OR
+            // (c) the oldest in-flight packet has expired
+            if (packetsSeen.canRecord(sendSeq))
+                return true;
+            if (udpAssumeSequential && packetsSeen.hasGaps())
+                return true;
+            if (msTick()-requestTime > udpResendTimeout)
+                return true;
+            return false;
+        }
+
+        void acknowledgeRequest(const IpAddress &returnAddress, unsigned _flowSeq, unsigned _sendSeq)
+        {
+            if (flowSeq==_flowSeq)
```

Comment:
should this also ignore earlier flowseq?  How out of order do we assume the packets could become?

Reply:
i think best to leave as-is, but may be worth revisiting.